src/librustdoc/core.rs

```diff
@@ -120,14 +120,20 @@ impl<'tcx> DocContext<'tcx> {
         r
     }
 
-    // This is an ugly hack, but it's the simplest way to handle synthetic impls without greatly
-    // refactoring either librustdoc or librustc_middle. In particular, allowing new DefIds to be
-    // registered after the AST is constructed would require storing the defid mapping in a
-    // RefCell, decreasing the performance for normal compilation for very little gain.
-    //
-    // Instead, we construct 'fake' def ids, which start immediately after the last DefId.
-    // In the Debug impl for clean::Item, we explicitly check for fake
-    // def ids, as we'll end up with a panic if we use the DefId Debug impl for fake DefIds
+    /// Create a new "fake" [`DefId`].
+    ///
+    /// This is an ugly hack, but it's the simplest way to handle synthetic impls without greatly
+    /// refactoring either `librustdoc` or [`rustc_middle`]. In particular, allowing new [`DefId`]s
```

Comment:
I don't think the directory is relevant here.
```suggestion
    /// refactoring either rustdoc or [`rustc_middle`]. In particular, allowing new [`DefId`]s
```

Reply:
Well `librustdoc` is also the name of the crate, but yeah I think it should be changed.

Reply:
The name of the crate is `rustdoc`. The name of the binary is `rustdoc-tool`.